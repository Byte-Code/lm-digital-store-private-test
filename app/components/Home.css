.container {
  display: flex;
  align-items: center;
  flex-direction: column;
  padding: 32px 0;
}

.container > img {
  width: 60%;
}

.container > h1 {
  margin-top: 32px;
  font-weight: lighter;
}
