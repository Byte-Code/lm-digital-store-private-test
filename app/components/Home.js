import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Map } from 'immutable';
import styled from 'styled-components';

import FamilyBadge from './FamilyBadge';

const FlexWrapper = styled.div`
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  width: 100%;
  &>div {
    width: calc(50% - 10px);
  }
`;

const GridWrapper = styled(FlexWrapper)`
  padding: 47px 40px 0;
`;

const DoubleVertical = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

const Banner = styled.div`
  background: url(${props => props.url}) center no-repeat border-box padding-box;
  height: 305px;
  font-size: 32px;
  padding: 30px 60px 47px 20px;
  color: #fff;
  font-family: LeroyMerlinSans Italic;
`;

const WorldTitle = styled.section`
  height: 140px;
  color: black;
  font-family: LeroyMerlinSans Bold;
  margin: 53px 0 30px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  &>div {
    font-size: 54px;
    width: calc(100% - 200px);
    &>span{
      color: #58c527;
    }
  }
  &>img {
    height: 128px;
    width: 213px;
  }
`;

export default class Home extends Component {
  static propTypes = {
    requestFetchWorld: PropTypes.func.isRequired,
    world: ImmutablePropTypes.map
  }

  static defaultProps = {
    world: Map()
  }

  componentDidMount() {
    this.props.requestFetchWorld();
  }

  render() {
    const { world } = this.props;

    if (world.isEmpty()) {
      return null;
    }

    const worldName = world.get('worldName');
    const families = world.get('families');
    const bannerText = world.getIn(['banner', 'text']);
    const bannerImg = world.getIn(['banner', 'image']);
    const trailingImage = world.get('trailingImage');

    return (
      <div>
        <GridWrapper>
          <div>
            <FamilyBadge family={families.get(0)} size="square-big" />
            <FlexWrapper>
              <DoubleVertical>
                <FamilyBadge family={families.get(1)} size="square-small" />
                <FamilyBadge family={families.get(2)} size="square-small" />
              </DoubleVertical>
              <FamilyBadge family={families.get(4)} size="vertical" />
            </FlexWrapper>
            <FamilyBadge family={families.get(7)} size="horizontal" />
          </div>
          <div>
            <FlexWrapper>
              <FamilyBadge family={families.get(6)} size="square-small" />
              <Banner url={bannerImg}>{bannerText}</Banner>
            </FlexWrapper>
            <FamilyBadge family={families.get(3)} size="square-big" />
            <FlexWrapper>
              <FamilyBadge family={families.get(9)} size="square-small" />
              <FamilyBadge family={families.get(8)} size="square-small" />
              <FamilyBadge family={families.get(5)} size="square-small" />
              <FamilyBadge family={families.get(10)} size="square-small" />
            </FlexWrapper>
          </div>
          <WorldTitle>
            <div>Esplora le soluzioni Leroy Merlin per
            <span>&nbsp;{worldName}</span></div>
            <img src="./assets/logo.png" alt="logo" />
          </WorldTitle>
        </GridWrapper>
        <img src={trailingImage} alt="alt" />
      </div>
    );
  }
}
