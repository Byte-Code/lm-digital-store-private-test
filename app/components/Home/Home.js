import React, { Component } from 'react';

export default class Home extends Component {
  render() {
    return (
      <div>
        <div>
          <img src="https://upload.wikimedia.org/wikipedia/commons/d/d4/Leroy_Merlin.svg" alt="logo" />
          <h1>Digital Store</h1>
        </div>
      </div>
    );
  }
}
