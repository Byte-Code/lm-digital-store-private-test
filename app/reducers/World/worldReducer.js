import { Map } from 'immutable';

export default function main(state = Map(), action) {
  switch (action.type) {
    case 'SUCCESS_FETCH_WORLD':
      return action.result;
    default:
      return state;
  }
}
