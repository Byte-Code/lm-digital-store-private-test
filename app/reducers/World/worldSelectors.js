export default function getWorld(state) {
  return state;
}
