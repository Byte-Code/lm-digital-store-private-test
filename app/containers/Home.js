import { connect } from 'react-redux';

import Home from '../components/Home';
import { requestFetchWorld } from '../actions/worldActions';
import getWorld from '../reducers/selectors';

const mapStateToProps = (state) => ({
  world: getWorld(state)
});

const mapDispatchToProps = {
  requestFetchWorld
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
