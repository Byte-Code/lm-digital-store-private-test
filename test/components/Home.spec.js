import React from 'react';
import { fromJS } from 'immutable';
import { shallow, mount } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

import Home from '../../app/components/Home';
import World from '../../mocks/world';

describe('Home', () => {
  const requestFetchWorld = jest.fn();

  it('should render null if world is empty', () => {
    const emptyWorld = fromJS({});
    const result = shallow(
      <Home
        world={emptyWorld}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should render properly if world is not empty', () => {
    const result = shallow(
      <Home
        world={fromJS(World)}
        requestFetchWorld={requestFetchWorld}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should call requestFetchWorld after mount', () => {
    mount(
      <MuiThemeProvider>
        <Home
          world={fromJS(World)}
          requestFetchWorld={requestFetchWorld}
        />
      </MuiThemeProvider>
    );
    expect(requestFetchWorld).toHaveBeenCalled();
  });
});
