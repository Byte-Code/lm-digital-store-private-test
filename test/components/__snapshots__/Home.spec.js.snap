// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render null if world is empty 1`] = `null`;

exports[`Home should render properly if world is not empty 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="logo"
      src="test-file-stub"
    />
  </styled.div>
  <styled.section>
    <h1>
      Esplora le soluzioni Leroy Merlin per
      <span>
        Â 
        Giardino e Terrazzo
      </span>
    </h1>
  </styled.section>
  <styled.div>
    <styled.div>
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Tavoli",
            image: "test-file-stub",
            categoryCode: "CAT656",
          }
        }
        size="square-big"
      />
      <styled.div>
        <FamilyBadge
          family={
            Immutable.Map {
              familyName: "Gazebo",
              image: "test-file-stub",
              categoryCode: "CAT944",
            }
          }
          size="square-small"
        />
        <FamilyBadge
          family={
            Immutable.Map {
              familyName: "Sedie e Poltrone",
              image: "test-file-stub",
              categoryCode: "CAT657",
            }
          }
          size="square-small"
        />
      </styled.div>
      <styled.div>
        <FamilyBadge
          family={
            Immutable.Map {
              familyName: "Panchine",
              image: "test-file-stub",
              categoryCode: "CAT945",
            }
          }
          size="square-small"
        />
        <FamilyBadge
          family={
            Immutable.Map {
              familyName: "Dondoli e Amache",
              image: "test-file-stub",
              categoryCode: "CAT659",
            }
          }
          size="square-small"
        />
      </styled.div>
    </styled.div>
    <styled.div>
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Ombrelloni e Basi",
            image: "test-file-stub",
            categoryCode: "CAT660",
          }
        }
        size="square-small"
      />
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Set di tavoli e sedie",
            image: "test-file-stub",
            categoryCode: "CAT655",
          }
        }
        size="horizontal"
      />
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Sdraio e Lettini",
            image: "test-file-stub",
            categoryCode: "CAT658",
          }
        }
        size="square-small"
      />
    </styled.div>
    <styled.div>
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Pergole",
            image: "test-file-stub",
            categoryCode: "CAT676",
          }
        }
        size="square-small"
      />
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Carport",
            image: "test-file-stub",
            categoryCode: "CAT4231",
          }
        }
        size="square-small"
      />
      <FamilyBadge
        family={
          Immutable.Map {
            familyName: "Divisori Giardino",
            image: "test-file-stub",
            categoryCode: "CAT4097",
          }
        }
        size="horizontal"
      />
    </styled.div>
    <TutorialButton>
      <styled.div
        url="test-file-stub"
      >
        Scopri le funzioni del Digital Store
        <ActionHelpOutline
          color="#fff"
          style={
            Object {
              "height": 65,
              "width": 65,
            }
          }
        />
      </styled.div>
    </TutorialButton>
  </styled.div>
  <styled.img
    alt="alt"
    src="test-file-stub"
  />
</styled.div>
`;
